Assets/SEE/UI/Window/PropertyWindow/PropertyWindow.cs

```diff
@@ -0,0 +1,845 @@
+        {
+            Dictionary<string, object> nestedDict = new();
+            foreach (KeyValuePair<string, object> pair in attributes)
+            {
+                AddNestedAttribute(nestedDict, pair.Key.Split('.'), pair.Value);
+            }
+            CreateNestedGroups(nestedDict);
+        }
+
+        /// <summary>
+        /// Inserts an attribute to the nested dictionary.
+        /// </summary>
+        /// <param name="dict">The nested dictionary.</param>
+        /// <param name="keys">The keys to add.</param>
+        /// <param name="value">The value of the attribute.</param>
+        private void AddNestedAttribute(Dictionary<string, object> dict, string[] keys, object value)
+        {
+            Dictionary<string, object> currentDict = dict;
+
+            for (int i = 0; i < keys.Length - 1; i++)
+            {
+                if (!currentDict.ContainsKey(keys[i]))
+                {
+                    currentDict[keys[i]] = new Dictionary<string, object>();
+                }
+                currentDict = (Dictionary<string, object>)currentDict[keys[i]];
+            }
+            currentDict[keys[^1]] = value;
+        }
+
+        /// <summary>
+        /// Creates the nested groups and attributes.
+        /// </summary>
+        /// <param name="dict">The nested dicitonary.</param>
+        /// <param name="groupName">The group name in which the object should be added.</param>
+        /// <param name="level">The hierarchy level.</param>
+        private void CreateNestedGroups(Dictionary<string, object> dict, string groupName = null, int level = 0)
+        {
+            List<KeyValuePair<string, object>> sortedList = dict.ToList();
+            sortedList = sortedList.OrderBy(kvp => kvp.Value is Dictionary<string, object> ? 1 : 0).ToList();
+            foreach (KeyValuePair<string, object> pair in sortedList)
+            {
+                if (pair.Value is Dictionary<string, object> nestedDict)
+                {
+                    DisplayGroup(pair.Key, new Dictionary<string, string>(), level, groupName);
+                    CreateNestedGroups(nestedDict, pair.Key, level + 1);
+                }
+                else
+                {
+                    groupName ??= "Header";
+                    DisplayAttributes(new Dictionary<string, string>() { { pair.Key, pair.Value.ToString() } },
+                        level, expandedItems.Contains(groupName), groupName);
+                }
+            }
+        }
+        #endregion
+
+        #region Sort
+        /// <summary>
+        /// Sorts the properties within the group.
+        /// </summary>
+        private void Sort()
+        {
+            if (contextMenu.Sorter.IsActive())
+            {
+                foreach (IEnumerable<GameObject> values in groupHolder.Values)
+                {
+                    List<GameObject> list = values.ToList();
+                    list.RemoveAll(x => !x.name.Contains("RowLine"));
```

Comment:
Since we are only consuming the values once, we can keep it as an `IEnumerable`:

```suggestion
                    values = values.Where(x => x.name.Contains("RowLine"));
```